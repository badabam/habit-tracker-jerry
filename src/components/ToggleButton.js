import React, { Component } from 'react'
import styled from 'react-emotion'

const StyledButton = styled('button')`
  width: 100%;
  height: 40px;
  background: ${props => (props.checked ? 'greenyellow' : '#eee')};
  border: 1px solid #333;
`

export default class ToggleButton extends Component {
  render() {
    return (
      <StyledButton checked={this.props.checked} onClick={this.props.onClick}>
        {this.props.text}
      </StyledButton>
    )
  }
}
