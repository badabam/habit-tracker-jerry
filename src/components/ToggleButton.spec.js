import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import ToggleButton from './ToggleButton'

describe('ToggleButton', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(<ToggleButton text="Foo" checked={false} />)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('toggles color correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ checked: true })
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
