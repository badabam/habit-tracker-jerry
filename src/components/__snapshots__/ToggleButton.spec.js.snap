// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  background: #ddd;
  border: 1px solid #ccc;
  border-radius: 3px;
}

<ToggleButton
  checked={false}
  text="Foo"
>
  <Styled(section)
    checked={false}
  >
    <section
      checked={false}
      className="emotion-0"
    >
      Foo
    </section>
  </Styled(section)>
</ToggleButton>
`;

exports[`ToggleButton toggles color correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  background: #ddd;
  border: 1px solid #ccc;
  border-radius: 3px;
}

<ToggleButton
  checked={false}
  text="Foo"
>
  <Styled(section)
    checked={false}
  >
    <section
      checked={false}
      className="emotion-0"
    >
      Foo
    </section>
  </Styled(section)>
</ToggleButton>
`;

exports[`ToggleButton toggles color correctly 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  background: skyblue;
  border: 1px solid #ccc;
  border-radius: 3px;
}

<ToggleButton
  checked={true}
  text="Foo"
>
  <Styled(section)
    checked={true}
  >
    <section
      checked={true}
      className="emotion-0"
    >
      Foo
    </section>
  </Styled(section)>
</ToggleButton>
`;
